.github/workflows/prbuild.yaml

```diff
@@ -106,6 +106,54 @@ jobs:
           LOAD_PREBUILT_IMAGE: "true"
         run: |
           make integration
+  upgrade:
+    runs-on: ubuntu-latest
+    needs: [build-image]
+    strategy:
+      matrix:
+        # use stable kubernetes_version values since they're included
+        # in the name of the GitHub Actions job, and we don't want to
+        # have to update branch protection rules every time we change
+        # a Kubernetes version number.
+        kubernetes_version: ["kubernetes:latest", "kubernetes:n-1", "kubernetes:n-2"]
+        # include defines an additional variable (the specific node
+        # image to use) for each kubernetes_version value.
+        include:
+          - kubernetes_version: "kubernetes:latest"
+            node_image: "docker.io/kindest/node:v1.21.1"
+          - kubernetes_version: "kubernetes:n-1"
+            node_image: "docker.io/kindest/node:v1.20.2"
+          - kubernetes_version: "kubernetes:n-2"
+            node_image: "docker.io/kindest/node:v1.19.7"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download image
+        uses: actions/download-artifact@v2
+        with:
+          name: image
+          path: image
+      - uses: actions/cache@v2
+        with:
+          # * Module download cache
+          # * Build cache (Linux)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: ${{ runner.os }}-${{ github.job }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ github.job }}-go-
+      - name: add deps to path
+        run: |
+          ./hack/actions/install-kubernetes-toolchain.sh $GITHUB_WORKSPACE/bin
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+      - name: upgrade tests
+        env:
+          NODEIMAGE: ${{ matrix.node_image }}
+          LOAD_PREBUILT_IMAGE: "true"
+          CONTOUR_UPGRADE_FROM_VERSION: v1.15.1
```

Comment:
Should add this to the release notes somewhere so we don't forget to change it.

Reply:
hmm yeah good point, i'll add it to the release-process.md doc to start?

Reply:
documenting for now sounds good. we might be able to compute this -- one way off the top of my head is to curl `https://github.com/projectcontour/contour/releases/latest`, follow the redirect, and grab the version from there

Reply:
Actually you already have code to compute it from git tags, do we even need to pass it explicitly here?

Reply:
ah yeah i forgot about that, it wasn't working in CI b/c of the way a PR branch is cloned but I can try to get it to work

Reply:
should be fixed now